code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DeploymentManagerImpl.java

```diff
@@ -268,12 +269,26 @@ public void doWithLockNoResult() {
 
     @Override
     public void activateGlobalServicesForHost(long accountId, long hostId) {
+
```

Comment:
By deferring the service.activate call, are we 100% sure that the action won't happen before the host is fully active? If some delay happens in host activation process for example